src/WebJobs.Script/Diagnostics/OpenTelemetry/OpenTelemetryConfigurationExtensions.cs

```diff
@@ -28,42 +28,34 @@ internal static class OpenTelemetryConfigurationExtensions
     {
         internal static void ConfigureOpenTelemetry(this ILoggingBuilder loggingBuilder, HostBuilderContext context, TelemetryMode telemetryMode)
         {
-            var connectionString = GetConfigurationValue(EnvironmentSettingNames.AppInsightsConnectionString, context.Configuration);
-            var (azMonConnectionString, credential, enableOtlp, enableAzureMonitor) = telemetryMode switch
-            {
-                // Initializing OTel services during placeholder mode as well to avoid the cost of JITting these objects during specialization.
-                // Azure Monitor Exporter requires a connection string to be initialized. Use placeholder connection string if in placeholder mode.
-                TelemetryMode.Placeholder => (
-                    "InstrumentationKey=00000000-0000-0000-0000-000000000000;",
-                    null,
-                    true,
-                    true),
-                _ => (
-                    connectionString,
-                    GetTokenCredential(context.Configuration),
-                    !string.IsNullOrEmpty(GetConfigurationValue(EnvironmentSettingNames.OtlpEndpoint, context.Configuration)),
-                    !string.IsNullOrEmpty(connectionString))
-            };
-
-            // If neither OTLP nor Azure Monitor is enabled, don't configure OpenTelemetry.
-            if (!enableOtlp && !enableAzureMonitor)
+            var otlpEndpoint = GetConfigurationValue(EnvironmentSettingNames.OtlpEndpoint, context.Configuration);
+            var azMonConnectionString = GetConfigurationValue(EnvironmentSettingNames.AppInsightsConnectionString, context.Configuration);
+
+            bool enableOtlp = !string.IsNullOrWhiteSpace(otlpEndpoint);
+            bool enableAzureMonitor = !string.IsNullOrWhiteSpace(azMonConnectionString);
+
+            TokenCredential credential = enableAzureMonitor ? GetTokenCredential(context.Configuration) : null;
+
+            // If placeholder mode is disabled and both OTLP and Azure Monitor are not enabled, avoid configuring OpenTelemetry.
+            if (!enableOtlp && !enableAzureMonitor && telemetryMode != TelemetryMode.Placeholder)
```

Comment:
Do we need the `telemetryMode` value check? If the endpoints are not configured, can't we just return/exit from here? (Also if the telemetryMode check is confusing to me. Should that be `==` ? (Because in placeholder mode, the value of `telemetryMode` is  `Placeholder`. Isn't it?).